examples/pauseable.rs

```diff
@@ -0,0 +1,98 @@
+    // responds to input events. When running == true, the application updates a stopwatch,
+    // tracking how much time has passed. The spacebar key pauses/resumes.
+    //
+    // TODO: Just let it run. The stopwatch time diverges from the total time since the start.
+    // TODO: Produce input events (by wiggling the mouse, for example). More than 10 updates per
+    // second happen.
+
+    // 10 updates per second
+    let update_frequency = Duration::from_millis(100);
+    let started_at = Instant::now();
+
+    let mut running = true;
+    let mut last_update = Instant::now();
+    let mut total = Duration::new(0, 0);
+
+    event_loop.run(move |event, _, control_flow| {
+        match event {
+            Event::WindowEvent {
+                event: WindowEvent::CloseRequested,
+                ..
+            } => *control_flow = ControlFlow::Exit,
+            Event::WindowEvent {
+                event:
+                    WindowEvent::KeyboardInput {
+                        input:
+                            KeyboardInput {
+                                virtual_keycode: Some(VirtualKeyCode::Space),
+                                state: ElementState::Pressed,
+                                ..
+                            },
+                        ..
+                    },
+                ..
+            } => {
+                window.request_redraw();
+                running = !running;
+                if running {
+                    let now = Instant::now();
+                    last_update = now;
+                    *control_flow = ControlFlow::WaitUntil(now + update_frequency);
+                    println!("- Resuming!");
+                } else {
+                    *control_flow = ControlFlow::Wait;
+                    println!("- Pausing");
+                }
+            }
+            Event::WindowEvent { .. } => {
+                println!("- Handling some input");
+                // Pretend that handling the event takes about 1ms
+                thread::sleep(Duration::from_millis(1));
+                window.request_redraw();
+            }
+            Event::RedrawRequested(_) => {
+                println!("  - Redrawing");
+                // Pretend that redrawing takes about 10ms
+                thread::sleep(Duration::from_millis(10));
+            }
+            Event::MainEventsCleared => {
+                if running {
+                    let dt = last_update.elapsed();
+                    total += dt;
+                    println!(
+                        "- Update: +{:?} for a total of {:?}. It's been {:?} since the very start",
+                        dt,
+                        total,
+                        started_at.elapsed()
+                    );
+                    // Pretend that updating the application takes about 20ms
+                    thread::sleep(Duration::from_millis(20));
```

Comment:
Won't `last_update` diverge by 20ms because the `let now = ...` is below this line?

Reply:
Ah, thanks! I'm not sure why it felt more natural to measure the last update after applying it. Switching the order seems to make this example work as expected.